client/src/api/block_builder/input_selection/new/requirement/foundry.rs

```diff
@@ -0,0 +1,45 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use super::{InputSelection, Requirement};
+use crate::{
+    block::output::{FoundryId, Output},
+    error::{Error, Result},
+    secret::types::InputSigningData,
+};
+
+/// Checks if an output is a foundry with foundry ID that matches the given one.
+pub(crate) fn is_foundry_with_id(output: &Output, foundry_id: &FoundryId) -> bool {
+    if let Output::Foundry(foundry) = output {
+        &foundry.id() == foundry_id
+    } else {
+        false
+    }
+}
+
+impl InputSelection {
+    /// Fulfills a foundry requirement by selecting the appropriate foundry from the available inputs.
+    pub(crate) fn fulfill_foundry_requirement(
+        &mut self,
+        foundry_id: FoundryId,
+    ) -> Result<(Vec<InputSigningData>, Option<Requirement>)> {
+        // Check if the requirement is already fulfilled.
+        if self
+            .selected_inputs
+            .iter()
+            .any(|input| is_foundry_with_id(&input.output, &foundry_id))
+        {
+            return Ok((Vec::new(), None));
```

Comment:
Maybe this is done somewhere else, but I can't see it at the moment
Shouldn't this return the controlling alias as requirement?

Reply:
This is solving the requirement, not detecting it, the controlling alias is found as requirement here https://github.com/iotaledger/iota.rs/blob/017ca875048d51d2138bf3daa6bfa96a742acfb6/client/src/api/block_builder/input_selection/new/requirement/mod.rs#L131

Reply:
For what do we have the option to return a Requirement then?

Reply:
Base token has to return a requirement, I just thought I would harmonise them in case they all need to return something at some point but it doesn't look like they do. I can remove.

Reply:
Whatever you think is better, was just curious

Reply:
Actually it's less error conversion if I keep it the way it is now